src/cluster.c

```diff
@@ -1332,16 +1332,18 @@ void addNodeToNodeReply(client *c, clusterNode *node) {
  * not finished their initial sync, in failed state, or are
  * otherwise considered not available to serve read commands. */
 int isNodeAvailable(clusterNode *node) {
+    /* We don't consider PFAIL here because it's not a reliable indicator
+     * for node available and we don't want clients to use it. */
```

Comment:
I know that we rejected pfail in the past, but i can add a point of view. From myself own perspective, if a node is pfail, it means that it is unavailable from myself view and we can mark it unavailable in myself response.

Reply:
I don't quite follow your comment. There isn't much harm in removing a temporarily unavailable node, but showing a dead node to clients is bad since they will timeout connecting to it.

Reply:
I think what @enjoy-binbin meant here is `PFAIL` state is not taken into consideration as it's just a self view and not yet confirmed by the quorum.

Reply:
yes, hpatro saying is right. Sorry about the confuse, bad english.

there are two comments, the one in the code, is somehow i took from the old PR, maybe the CLUSTER SHARDS one, back then, we think a pfail node is available, so in here, we only skip fail node, the clusterNodeIsFailing one.

and in here, the comment i added as a github review comment, is saying,  a pfail node may be unavailable in myself's view. The decision we made in the past in cluster shard was that the pfail node was available so we won't skip it. But if we look at it from myself own perspective, if a node pfails, can we assume that myself think it is unavailable?